src/test/amount_tests.cpp

```diff
@@ -111,6 +111,19 @@ BOOST_AUTO_TEST_CASE(CFeeRateNamedConstructorsTest)
     BOOST_CHECK(CFeeRate::FromSatB(CAmount(123456789)) == CFeeRate(1234, 1000));
 }
 
+BOOST_AUTO_TEST_CASE(CFeeRateIsZeroTest)
+{
+    BOOST_CHECK(CFeeRate(0).IsZero());
+    BOOST_CHECK(!CFeeRate(1000).IsZero());
+    BOOST_CHECK(!CFeeRate(-1000).IsZero());
+    BOOST_CHECK(CFeeRate(CAmount(0), 0).IsZero());
+    BOOST_CHECK(CFeeRate(CAmount(1), 1001).IsZero());
+    BOOST_CHECK(CFeeRate::FromBtcKb(CAmount(0)).IsZero());
+    BOOST_CHECK(!CFeeRate::FromBtcKb(CAmount(1)).IsZero());
+    BOOST_CHECK(CFeeRate::FromSatB(CAmount(0.000)).IsZero());
```

Comment:
in commit 83fb6101f0768e8f540e187caa2679c9f0b98c20: CAmount can't represent floats. This would be a compile error, if you didn't use a c-style-narrowing-cast

Reply:
For reference:

```
test/amount_tests.cpp:123:44: error: type 'double' cannot be narrowed to 'CAmount' (aka 'long') in initializer list [-Wc++11-narrowing]
    BOOST_CHECK(CFeeRate::FromSatB(CAmount{0.000}).IsZero());
                                           ^~~~~
```

Reply:
changed to 0